kotest-extensions/kotest-extensions-junit5extensions/src/jvmMain/kotlin/io/kotest/extensions/junit5/KotestExtensionContext.kt

```diff
@@ -0,0 +1,104 @@
+package io.kotest.extensions.junit5
+
+import io.kotest.core.spec.Spec
+import io.kotest.core.spec.description
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.extension.ExtensionContext
+import org.junit.jupiter.api.extension.TestInstances
+import java.lang.reflect.AnnotatedElement
+import java.lang.reflect.Method
+import java.util.Optional
+import java.util.function.Function
+
+class KotestExtensionContext(private val spec: Spec) : ExtensionContext {
+
+   override fun getParent(): Optional<ExtensionContext> = Optional.empty()
+   override fun getRoot(): ExtensionContext = this
+
+   override fun getUniqueId(): String = spec::class.description().id()
+
+   override fun getDisplayName(): String = spec::class.description().fullName()
+
+   override fun getTags(): MutableSet<String> = spec.tags().map { it.name }.toMutableSet()
+
+   override fun getElement(): Optional<AnnotatedElement> = Optional.empty()
+
+   override fun getTestClass(): Optional<Class<*>> = Optional.of(spec::class.java)
+   override fun getTestMethod(): Optional<Method> = Optional.empty()
+
+   override fun getTestInstanceLifecycle(): Optional<TestInstance.Lifecycle> =
+      Optional.of(TestInstance.Lifecycle.PER_CLASS)
+
+   override fun getTestInstance(): Optional<Any> = Optional.of(spec)
+   override fun getTestInstances(): Optional<TestInstances> = Optional.of(KotestTestInstances(spec))
+
+   override fun getExecutionException(): Optional<Throwable> = Optional.empty()
+   override fun getConfigurationParameter(key: String?): Optional<String> = Optional.empty()
+
+   override fun publishReportEntry(map: MutableMap<String, String>?) {}
+
+   override fun getStore(namespace: ExtensionContext.Namespace): ExtensionContext.Store {
+      return ExtensionStore(namespace)
+   }
+}
+
+class KotestTestInstances(private val instance: Spec) : TestInstances {
+   override fun getInnermostInstance(): Any = instance
+   override fun getEnclosingInstances(): MutableList<Any> = mutableListOf(instance)
+   override fun getAllInstances(): MutableList<Any> = mutableListOf(instance)
+   override fun <T : Any?> findInstance(requiredType: Class<T>?): Optional<T> = Optional.of(instance as T)
```

Comment:
I think here as well we can potentially get a class cast exception.

Reply:
Thanks, fixed.